SetupDataPkg/ConfDataSettingProvider/UnitTest/ConfDataSettingProviderUnitTest.c

```diff
@@ -0,0 +1,1955 @@
+                                firmware volume that
+                                contains the matching FFS section does not allow reads.
+**/
+EFI_STATUS
+EFIAPI
+GetSectionFromAnyFv (
+  IN CONST  EFI_GUID          *NameGuid,
+  IN        EFI_SECTION_TYPE  SectionType,
+  IN        UINTN             SectionInstance,
+  OUT       VOID              **Buffer,
+  OUT       UINTN             *Size
+  )
+{
+  VOID  *ret_buf;
+
+  assert_memory_equal (NameGuid, PcdGetPtr (PcdConfigPolicyVariableGuid), sizeof (EFI_GUID));
+  assert_int_equal (SectionType, EFI_SECTION_RAW);
+  assert_non_null (Buffer);
+  assert_non_null (Size);
+
+  ret_buf = (VOID *)mock ();
+  if (ret_buf != NULL) {
+    *Size   = (UINTN)mock ();
+    *Buffer = AllocateCopyPool (*Size, ret_buf);
+  } else {
+    return EFI_NOT_FOUND;
+  }
+
+  return EFI_SUCCESS;
+}
+
+/**
+Registers a Setting Provider with the System Settings module
+
+@param  This                 Protocol instance pointer.
+@param  Provider             Provider pointer to register
+
+@retval EFI_SUCCESS          The provider registered.
+@retval ERROR                The provider could not be registered.
+
+**/
+EFI_STATUS
+EFIAPI
+MockDfciRegisterProvider (
+  IN DFCI_SETTING_PROVIDER_SUPPORT_PROTOCOL  *This,
+  IN DFCI_SETTING_PROVIDER                   *Provider
+  )
+{
+  assert_non_null (This);
+  assert_non_null (Provider);
+
+  DEBUG ((DEBUG_INFO, "%a Registering ID %a\n", __FUNCTION__, Provider->Id));
+
+  check_expected (Provider->Id);
+
+  return EFI_SUCCESS;
+}
+
+// Mocked version of setting provider register protocol instance.
+DFCI_SETTING_PROVIDER_SUPPORT_PROTOCOL  mMockDfciSetting = {
+  .RegisterProvider = MockDfciRegisterProvider
+};
+
+/**
+  Creates and returns a notification event and registers that event with all the protocol
+  instances specified by ProtocolGuid.
+
+  This function causes the notification function to be executed for every protocol of type
+  ProtocolGuid instance that exists in the system when this function is invoked. If there are
+  no instances of ProtocolGuid in the handle database at the time this function is invoked,
+  then the notification function is still executed one time. In addition, every time a protocol
+  of type ProtocolGuid instance is installed or reinstalled, the notification function is also
+  executed. This function returns the notification event that was created.
+  If ProtocolGuid is NULL, then ASSERT().
+  If NotifyTpl is not a legal TPL value, then ASSERT().
+  If NotifyFunction is NULL, then ASSERT().
+  If Registration is NULL, then ASSERT().
+
+
+  @param  ProtocolGuid    Supplies GUID of the protocol upon whose installation the event is fired.
+  @param  NotifyTpl       Supplies the task priority level of the event notifications.
+  @param  NotifyFunction  Supplies the function to notify when the event is signaled.
+  @param  NotifyContext   The context parameter to pass to NotifyFunction.

```

Comment:
Maybe include the optional tag here as well.

Reply:
Udpated.